src/rdkafka_lz4.c

```diff
@@ -272,13 +278,16 @@ rd_kafka_lz4_decompress (rd_kafka_broker_t *rkb, int proper_hc, int64_t Offset,
                         size_t extra = RD_MAX(outlen * 3 / 4, 1024);
 
                         rd_atomic64_add(&rkb->rkb_c.zbuf_grow, 1);
+                        rd_bool_t succeeded;
+                        rd_wait_mem_space(extra, succeeded);
 
-                        if (!(tmp = rd_realloc(out, outlen + extra))) {
+                        if (succeeded == rd_false || !(tmp = rd_realloc(out, outlen + extra))) {
```

Comment:
They correctly use realloc here, also check the resizing strategy - they grow by 1.75 factor, probably we should do the same in zstd codepath

Reply:
Though x2 factor should be friendlier to allocators??

Reply:
I was planing to  reduce it to 2 and higher the initial buffer size